src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerActionExecutor.cs

```diff
@@ -58,6 +67,30 @@ private static Task<object> Convert<T>(object taskAsObject)
                 actionMethodInfo.DeclaringType);
         }
 
+        public static Task<object> ExecuteAsync(
+            ObjectMethodExecutor actionMethodExecutor,
+            object instance,
+            object[] orderedActionArguments)
+        {
+            object invocationResult = null;
+            try
+            {
+                invocationResult = actionMethodExecutor.Execute(instance, orderedActionArguments);
+            }
+            catch (Exception actionExecutionException)
+            {
+                // Capturing the exception and the original callstack and rethrow for external exception handlers.
+                var exceptionDispatchInfo = ExceptionDispatchInfo.Capture(actionExecutionException);
+                exceptionDispatchInfo.Throw();
+            }
```

Comment:
This try/catch shouldn't be needed for the expressions. This was only needed for the reflection case because the exception was wrapped by `TargetInvocationException`